### DISCUSSION TITLE
When to use Valtio, when to use Jotai

### DISCUSSION THREAD

**Discussion by opticode** [pmndrs/valtio#128] [Q&A] [answered]
@dai-shi as the creator of multiple state management libraries, I am looking for some real-world practical advice on when you would consider using Valtio in a project vs using Jotai in a project. I know this may be a hard question but was hoping for some generalizations :)

For context, my (modern react) app will have lots of inter-dependent nested data, and interactive user interaction with that data. My experiments with Valtio so far has been amazing, but I'm wondering if I'll run into issues with more complexity in my app, and if I should also experiment with Jotai.

Any advice would be greatly appreciated!

**Comment by dai-shi** [comment, marked as answer]
There are coding style differences, maturity difference, popularity differences, and some technical/internal differences. But, let's try not to answer "it depends" this time.

Assuming jotai will hit v1 hopefully soon, my _general_ recommendation is jotai over valtio, _if_ people are considering two as options.
This is because the mental model of jotai is so much closer to that of react.
Like people report issues as they expect jotai to work in a certain way, and we fix it when it isn't aligned.
There are some exceptions, but it generally works as expected by people familiar with react.
I would expect if two developers used jotai for the same project, their usages should become similar.
Again, there are exceptions as jotai is very flexible and accepts various patterns.

Valtio, on the other hand, is magical, wow, it just works, with less code, just JS, various ways to implement action handlers, unopinionated like zustand.

Let me refer one of my tweets: https://twitter.com/dai_shi/status/1343694177884721152

> I'm wondering if I'll run into issues with more complexity in my app

There are many aspects to hit some unknown difficulties, but my gut feeling is that the possibility of such troubles would be somewhat equal between valtio and jotai on average. No evidence. 😎

**Reply by imranaalam** [reply]
what is better in case of vue? Valtio or Jotai?
is this the reason to prefer Jotai; as its more documented and more inturtive for react.
is Valtio better for Vue; as it starts from 'V' ;)

**Reply by dai-shi** [reply]
I'm not a vue user, but as far as I heard, valtio is very close to vue conceptually. So, if you want vue-like experience in react, pick valtio. If you look for something different with react, jotai might be fun.

**Reply by rpggio** [reply]
I've always had a problem with React being the host for all application state/effects. It never works well with routing, sockets, etc. because React is just a DOM library. Having all effects triggered by component mounting is like donning a hat to make your car start.
Definitely going to invest in Valtio. 
~I think all I lack is a query utility. Something like `useQuery()` which uses Valtio proxies for query state.~ I see the  [promise handling](https://github.com/pmndrs/valtio#suspend-your-components) support.
